appveyor.yml

```diff
@@ -20,16 +20,45 @@ install:
 
     choco install Firefox | out-null
 
-- cmd: >-
-    node --version
-
     npm --version
 
-    npm install
-
+    npm i -s
 build: off
 
 # NOTE: We should prerun Firefox to avoid "Refresh Firefox" notification in the tests window.
 test_script:
-- cmd: start /D "C:\Program Files (x86)\Mozilla Firefox" firefox.exe -new-window https://google.com
-- cmd: npm test
+- cmd: >-
+     echo %APPVEYOR_REPO_BRANCH%
+
+     echo %GULP_TASK%
+
+     start /D "C:\Program Files (x86)\Mozilla Firefox" firefox.exe -new-window http://example.com/
+
+     npm test
+
+for:
+-
+  branches:
+    except:
+      - master
+
+  notifications:
+  - provider: Email
+    to:
+      - mikhail.losev@devexpress.com
+    on_build_success: false
+    on_build_status_changed: false
+
+  test_script:
```

Comment:
I'm interested if this script can be simplified by overriding only the `GULP_TASK` environment variable. E.g.:
```
environment:
  GULP_TASK: "test-functional-local"
  NODEJS_VERSION: "stable"

for:
-
  branches:
    only:
      - master

  notifications:
  - provider: Email
    to:
      - mikhail.losev@devexpress.com
    on_build_success: false
    on_build_status_changed: false

  environment:
    GULP_TASK: "test-functional-local test-functional-legacy-local"

...

install:
  ...
     npm i -g gulp-cli

... 

test_script:
- cmd: >-
     echo %APPVEYOR_REPO_BRANCH%
     echo %GULP_TASK%
     start /D "C:\Program Files (x86)\Mozilla Firefox" firefox.exe -new-window http://example.com/
     for %g in (%GULP_TASK%) do gulp %g
```